### PR TITLE
Fix #150: Lowfi text input interaction view part 1

### LINKED ISSUE #150: UI Structure: TextInputInteractionView
This is tracking implementing the initial UI structure of the text input interaction being fully implemented in #28.

### PR DESCRIPTION
**Pull request by nikitamarysolomanpvt**
Replicated from #190

## Explanation
Part 1: This PR contains Text input view and it Fix  #150  .

## Reference Mock
https://xd.adobe.com/spec/e2239cf4-9cde-4c08-5296-25316c1f0a14-9412/screen/4d4e4bd8-3755-41fc-bf83-3dcff5d5eee1/Lesson-1-a-ii-Exploration-Player-Text-Input-No-Ans
https://xd.adobe.com/spec/e2239cf4-9cde-4c08-5296-25316c1f0a14-9412/screen/eeed0f6c-54ca-4371-ab23-5199abd87f28/Lesson-1-a-ii-Exploration-Player-Text-Input-Keyboa

## Reference Design Doc
https://docs.google.com/document/d/1QigNdZeMBy-JjHW26QH4qT0UtCyzfvW0xEaStSdY9gM/edit?usp=sharing
## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is **assigned** to an appropriate reviewer.

### CODE REVIEW

File: app/src/main/java/org/oppia/app/activity/InputInteractionViewTestActivity.kt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import org.oppia.app.R
+import org.oppia.app.customview.inputInteractionView.NumericInputInteractionView
+import org.oppia.app.customview.inputInteractionView.TextInputInteractionView
+
+private const val KEY_NUMERIC_INPUT_ID = "NUMERIC_INPUT_ID"
+private const val KEY_TEXT_INPUT_ID = "TEXT_INPUT_ID"
+
+/**
+ * This is a dummy activity to test input interaction views.
+ * it contains [NumericInputInteractionView] and [TextInputInteractionView].
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/activity/InputInteractionViewTestActivity.kt:14]: Nit: First character capital. `It contains ...`

File: app/src/main/java/org/oppia/app/activity/InputInteractionViewTestActivity.kt [hunk 2]
```diff
@@ -0,0 +1,39 @@
+package org.oppia.app.activity
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/activity/InputInteractionViewTestActivity.kt:1]: Should we keep this file in this package? I suggest shifting it to `customview` or `state`.
Basically, wherever the implementation is these test-activities should be in that only.

File: app/src/main/java/org/oppia/app/activity/InputInteractionViewTestActivity.kt [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+ */
+class InputInteractionViewTestActivity : AppCompatActivity() {
+  private var numericInput: String = ""
+  private var textInput: String = ""
+  private lateinit var contentNumericComponent: NumericInputInteractionView
+  private lateinit var contentTextComponent: TextInputInteractionView
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_numeric_input_interaction_view_test)
+    contentNumericComponent = findViewById(R.id.test_number_input_interaction_view) as NumericInputInteractionView
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/activity/InputInteractionViewTestActivity.kt:24]: Prefer writing it this way:

`contentNumericComponent = findViewById(R.id.test_number_input_interaction_view)`

Ditto for next line

File: app/src/main/java/org/oppia/app/activity/InputInteractionViewTestActivity.kt [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    contentTextComponent = findViewById(R.id.test_text_input_interaction_view) as TextInputInteractionView
+    if (savedInstanceState != null) {
+      numericInput = savedInstanceState.getString(KEY_NUMERIC_INPUT_ID)
+      textInput = savedInstanceState.getString(KEY_TEXT_INPUT_ID)
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/activity/InputInteractionViewTestActivity.kt:28]: Either make the `textInput` support null by changing line 18 to 

`private var textInput: String? = ""`

or else check that the `savedInstanceState.getString(KEY_TEXT_INPUT_ID)`  is not null.

Comment (rt4914) [reply app/src/main/java/org/oppia/app/activity/InputInteractionViewTestActivity.kt:28]: Ditto for numericInput

File: app/src/main/java/org/oppia/app/customview/inputInteractionView/TextInputInteractionView.kt [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  /** This function contains default attributes of [TextInputInteractionView]. */
+  private fun attributes() {
+    setBackgroundResource(R.drawable.edit_text_background)
+    val paddingPixel = context.resources.getDimension(R.dimen.padding_8)
+    val density = resources.displayMetrics.density
+    val paddingDp = (paddingPixel * density).toInt()
+    setPadding(paddingDp, paddingDp, paddingDp, paddingDp)
+    hint = placeholder
+    this.setEditTextMaxLength(length = 200)
+    setLongClickable(false)
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/customview/inputInteractionView/TextInputInteractionView.kt:35]: Update this to `isLongClickable = false`

File: app/src/main/java/org/oppia/app/customview/inputInteractionView/TextInputInteractionView.kt [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  }
+
+  private fun setEditTextMaxLength(length: Int) {
+    val filterArray = arrayOfNulls<InputFilter>(1)
+    filterArray[0] = InputFilter.LengthFilter(length)
+    filters = filterArray
+  }
+
+  override fun getPendingAnswer(): InteractionObject {
+    if (text.isEmpty()) {
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/customview/inputInteractionView/TextInputInteractionView.kt:45]: Simplify this to:

`return if (text.isEmpty()) {
      InteractionObject.newBuilder().build()
    } else {
      InteractionObject.newBuilder().setReal(text.toString().toDouble()).build()
    }`


As suggested earlier, prefer solving as many warnings as possible in Android Studio itself.

File: app/src/sharedTest/java/org/oppia/app/activity/InputInteractionViewTestActivityTest.kt [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+import org.junit.After
+import org.oppia.app.customview.inputInteractionView.NumericInputInteractionView
+import org.oppia.app.customview.inputInteractionView.TextInputInteractionView
+
+/**
+ * Tests for [InputInteractionViewTestActivity].
+ * it tests [NumericInputInteractionView] and [TextInputInteractionView].
+ */
+@RunWith(AndroidJUnit4::class)
+class InputInteractionViewTestActivityTest {
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/activity/InputInteractionViewTestActivityTest.kt:33]: Can only check these test-cases once @jamesxu0 PR on #241 is merged in develop.

**Review by rt4914 — CHANGES_REQUESTED** [review]
PTAL @nikitamarysolomanpvt

**Comment by nikitamarysolomanpvt** [comment]
Duplicate of this is #246 created due to change in approach

**Pull request closed without merging**